cpp/src/arrow/filesystem/s3fs.h

```diff
@@ -62,10 +75,18 @@ struct ARROW_EXPORT S3Options {
   void ConfigureAnonymousCredentials();
 
   /// Configure with explicit access and secret key.
-  void ConfigureAccessKey(const std::string& access_key, const std::string& secret_key);
+  void ConfigureAccessKey(const std::string& access_key, const std::string& secret_key,
+                          const std::string& session_token = "");
+
+  /// Configure with credentials from an assumed role.
+  void ConfigureAssumeRoleCredentials(
+      const std::string& role_arn, const std::string& session_name = "",
+      const std::string& external_id = "", int load_frequency = 900,
+      const std::shared_ptr<Aws::STS::STSClient>& stsClient = nullptr);
```

Comment:
`cpplint` does not like this, but I am just mirroring the default value used by the SDK.  Should I suppress the violation or is there indeed a better practice?  I'm a C++ neophyte so pointers (ha!) appreciated.

Reply:
I inferred this is something to do with compliance with C++/CLI; replaced with existing NULLPTR macro.